### ISSUE TITLE
Import NS Records with without any change in TTL

### ISSUE THREAD

**Issue by Subramanian-ERS** [StackExchange/dnscontrol#1417]
I could see that when I am trying to add a new DNS record in an already existing hosted zone, the output of the preview command indicates that the TTL of the NS records are being changed to 300. I have not added any NS records in my dnsconfig.js file.

While I understand that we can set the TTL using the NAMESERVER_TTL option, is there a way to ask DNSControl to maintain the existing TTL of the NS records?

**Comment by tlimoncelli** [comment]
Hi!  Thanks for pointing this out!

Yes, I agree it should be automatic unless you specify a NAMESERVER_TTL or NAMESERVER() item.

I suspect this might be a provider-specific issue.  Are you seeing this with certain providers or all providers?

Can you send a (redacted) version of dnsconfig.js? (just the D() for this domain and the related registrar and dns providers).

Thanks!

**Comment by Subramanian-ERS** [comment]
Hi @tlimoncelli, I am not using any provider other than Route53 for now.

This is the sample dnsconfig.js file:

var aws = NewDnsProvider("aws", "ROUTE53");
var REG_CHANGEME = NewRegistrar("ThirdParty", "NONE");
D("example.com", REG_CHANGEME,
	DnsProvider(aws),
	R53_ALIAS('@', 'A', 'example.elb.aws.com.', R53_ZONE('ABCD')),
	//NAMESERVER('ns-1.awsdns-1.org.'),
	//NAMESERVER('ns-2.awsdns-2.com.'),
	//NAMESERVER('ns-3.awsdns-3.co.uk.'),
	//NAMESERVER('ns-4.awsdns-4.net.'),
	CNAME('*', 'example.com.', TTL(60)),
)

**Comment by tlimoncelli** [comment]
Hi there!

I did some checking.  Yes, the TTLs must be set manually with `NAMESERVER_TTL()`.   DNSControl does not set the TTL automatically.  I'll tag this issue as a feature request.

ProTip: you can set `NAMESERVER_TTL()` for many domains by declaring `DEFAULT()` before any `D()` statements.

```
DEFAULTS(
    NAMESERVER_TTL(172800),
    END);
```

**Comment by tlimoncelli** [comment]
I haven't tried this yet but. Have you tried using the `IGNORE_TARGET()` function?

Maybe something like this?

```
    IGNORE_TARGET("**.example.com", "NS")
```

**Comment by Subramanian-ERS** [comment]
Hi @tlimoncelli,

I tried using IGNORE_TARGET, but it did not work. According to the documentation page, seems like it would work only for CNAME records.

**Comment by tlimoncelli** [comment]
Drat.  Oh well, it was worth a try! 

I looked at how we do this at Stackoverflow and we just manually set the TTL to match R53.  Once it is correct, that rarely changes.

**Comment by Subramanian-ERS** [comment]
Right. We will also manually configure it to have the same value for now though I would love this feature request to be implemented. 

But I will close it for now. Thanks.
